### PR TITLE
Update to Gradle 8.4 and add Java 21 to the matrixes

### PR DESCRIPTION
**Pull request by timyates**

### CODE REVIEW

File: .github/workflows/graalvm-latest.yml [hunk 1]
```diff
@@ -34,7 +34,9 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       max-parallel: 6
-      matrix: ${{ fromJson(needs.build_matrix.outputs.matrix) }}
+      matrix:
+        java: ['17', '21']
```
Comment (msupic) [inline .github/workflows/graalvm-latest.yml:38]: Do we need to run graalvm tests against both versions since this would duplicate number of slow native tests for each project but it would be especially painful for projects with a large number of those tests like `micronaut-sql` and `micronaut-data`?

Comment (timyates) [reply .github/workflows/graalvm-latest.yml:38]: I would say probably yes...

We want to be testing Micronaut native compilation against LTS JDK releases, and I guess this includes Graal JDKs

@sdelamo for micronaut-data this will double the number of GraalVM latest actions from 17 (currently)  to 34

Comment (sdelamo) [reply .github/workflows/graalvm-latest.yml:38]: I think we should

**Review by sdelamo — APPROVED** [review]
